pyproject.toml

```diff
@@ -78,6 +78,7 @@ dev = [
     "pytest-cov",
     "coverage",
     "bump-my-version",
+    "jsonschema",
```

Comment:
Maybe I missed something, but I don't see `jsonschema` being used anywhere. I think you are doing your validation with a function from `pystac`, is that right?

Reply:
Thanks for this @huard! 

About the version of the Datacube extension, how does a particular version of the extension get picked up? We never specify a version to use while applying the extension. Do different versions of `pystac` include different versions of the extension? When `add_CMIP6.py` runs on my machine, `pystac` applies version 2.0.0.

Reply:
Okay, so to answer my own question, looks like different versions of `pystac` come with different versions of extensions. Specifically, for data cube, `pystac 1.8.3` comes with 2.0.0 and `pystac 1.9.0` comes with 2.2.0.

I think that in `test_cmip6_datacube.py`, for the validation schema, we should use the schema that is directly referenced in the `stac_extensions` property of the item being validated, rather than use a a [specific hard-coded version](https://github.com/crim-ca/stac-populator/pull/33/commits/93af1ac111e56070d87c7a241488724b8dfbaf68). This way, there would be no mis-match between the schema version used while creating the item (this directly comes from the version of `pystac` used) and the schema version used for validation.

Reply:
I think it's required by pystac validation, but it may also be a left over from previous experimentation, not 100% sure. 

Yes, that's what I learned when trying to understand why the CI failed while it ran fine on my end. 

Yes, that makes a lot of sense.

Reply:
The test now validates against the version used. If it's 2.2, it gets it from the repo, otherwise it will fetch it from the net. I find it useful when debugging to have a hard-copy of the schema nearby, but we could ditch it at some point.